repos/system_upgrade/common/actors/opensshpermitrootlogincheck/actor.py

```diff
@@ -0,0 +1,153 @@
+from leapp import reporting
+from leapp.actors import Actor
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.actor.opensshpermitrootlogincheck import global_value, semantics_changes
+from leapp.libraries.common.config.version import get_source_major_version
+from leapp.libraries.stdlib import api
+from leapp.models import OpenSshConfig, Report
+from leapp.reporting import create_report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+COMMON_REPORT_TAGS = [
+    reporting.Tags.AUTHENTICATION,
+    reporting.Tags.SECURITY,
+    reporting.Tags.NETWORK,
+    reporting.Tags.SERVICES
+]
+
+COMMON_RESOURCES = [
+    reporting.RelatedResource('package', 'openssh-server'),
+    reporting.RelatedResource('file', '/etc/ssh/sshd_config')
+]
+
+
+class OpenSshPermitRootLoginCheck(Actor):
+    """
+    OpenSSH no longer allows root logins with password.
+
+    Check the values of PermitRootLogin in OpenSSH server configuration file
+    and warn about potential issues after upgrade to the next major version of RHEL.
+
+    The RHEL8 still provided default configuration that allowed root logins,
+    which can lead to possible unwanted changes during the upgrade
+    """
+    name = 'openssh_permit_root_login'
+    consumes = (OpenSshConfig, )
+    produces = (Report, )
+    tags = (ChecksPhaseTag, IPUWorkflowTag, )
+
+    def process(self):
+        openssh_messages = self.consume(OpenSshConfig)
+        config = next(openssh_messages, None)
+        if list(openssh_messages):
+            api.current_logger().warning('Unexpectedly received more than one OpenSshConfig message.')
+        if not config:
+            raise StopActorExecutionError(
+                'Could not check openssh configuration', details={'details': 'No OpenSshConfig facts found.'}
+            )
+
+        if get_source_major_version() == '7':
+            self.process7to8(config)
+        elif get_source_major_version() == '8':
+            self.process8to9(config)
+        else:
+            api.current_logger().warning('Unknown source major version: {} (expecting 7 or 8)'
+                                         .format(get_source_major_version()))
+
+    def process7to8(self, config):
+        # When the configuration does not contain the PermitRootLogin directive and
```

Comment:
@Jakuje do we not consider a commented PermitRootLogin directive anymore for the inhibitor?

Reply:
I do not think so. Why would we?

Reply:
@Jakuje 
Well, currently it works like that (in master). I was under impression that if the config file was modified and there is not explicitly set "PermitrootLogin", we trigger the inhibitor. Even if "PermitrootLogin" is commented, not removed. Did I get it wrong?

Reply:
Yes. That is kept in 7to8. If PermitRootLogin is commented or removed, it is like it would not be there at all (it should be a job of the parser though). This part of code is just moved from repos/system_upgrade/el7toel8/actors/opensshpermitrootlogincheck/actor.py:44 so it should not change on 7to8 execution.